pool.go

```diff
@@ -238,11 +238,7 @@ func (pool *Pool) Add(f func(...interface{}) interface{}, args ...interface{}) {
 }
 
 func (pool *Pool) getNextJobId() uint64 {
-	pool.next_job_id_mutex.Lock()
-	job_id := pool.next_job_id
-	pool.next_job_id++
-	pool.next_job_id_mutex.Unlock()
-	return job_id
+	return atomic.AddUint64(&pool.next_job_id, 1)
```

Comment:
This starts the IDs from 1 instead of 0.

Why rely on Go's implicit scheduler yielding suspension during ASM code execution, instead of explicit locks? This is what AddUint64 looks like on amd64: https://github.com/golang/go/blob/release-branch.go1.10/src/runtime/internal/atomic/asm_amd64.s#L91

> i still think atomic is better, simplify and effective

Explicit is better than implicit, when you are trying to make the code readable.

Reply:
Mutex is also based on atomic package.
```
func (m *Mutex) Lock() {
	// Fast path: grab unlocked mutex.
	if atomic.CompareAndSwapInt32(&m.state, 0, mutexLocked) {
		if race.Enabled {
			race.Acquire(unsafe.Pointer(m))
		}
		return
	}
        .........
```
Use atomic.AddUint64 directly can get more performance.

Reply:
OK, have your atomic increment.